src/mixins/numeral-formatable.js

```diff
@@ -0,0 +1,267 @@
+          this.blockSize[param.j] && this.blockSeparator[param.i] && param.masked.push(digit)
+        } else param.masked.push(digit)
+      }
+
+      if (param.integer) {
+        if (!this.blockSize[param.j]) return false
+        else if (!this.blockSeparator[param.i]) param.masked.push(digit)
+        else if (!(++param.length % this.blockSize[param.j])) {
+          param.masked.push(this.oneChar(this.blockSeparator[param.i], ' '))
+          param.i > 0 && param.i--
+          param.j > 0 && param.j--
+          param.length = 0
+        }
+      } else if (digit === '.') {
+        param.masked.push(this.decimal)
+        param.integer = true
+      }
+
+      return true
+    },
+    maskedNumeral (masked) {
+      if (!masked.length) masked.splice(0, 0, '0')
+      else if (this.blockSeparator.includes(masked[0])) masked.shift()
+
+      const result = this.addPrefixSuffix(masked.join(''))
+
+      return masked[0] === '.' ? '0' + result : result
+    },
+    numeralZero () {
+      const zero = 0
+      this.sign = '+'
+      return this.addPrefixSuffix(zero.toFixed(this.precision).toString().replace('.', this.decimal))
+    },
+    oneChar (char, defaultChar = '.') {
+      return char ? char[0] : defaultChar
+    },
+    // TODO: Rework this restrictive function to
+    // make -ve as a valid value in order
+    // to support special meaning/feature
+    absVal (value) {
+      const parsed = parseInt(value, 10)
+
+      if (isNaN(parsed)) return 0
+      return Math.abs(parsed)
+    },
+    adjustNumeralCaret (selection, text) {
+      // Caret is at prefix
+      if (selection === 0) return this.numeralPrefix.length + this.sign.length
+
+      // Whether selection !== 0 (may happen when block size === 0) or it's an
+      // integer or a fractional where abs(value) >= 10
+      if (selection !== text.length && (!this.precision ||
+        (String(this.lazyValue).length > this.precision + (this.sign ? 3 : 2)))) {
+        return selection
+      }
+
+      // Caret is at suffix
+      const suffixPosition = this.numeralSuffix ? text.lastIndexOf(this.numeralSuffix) : 0
+      if (suffixPosition > 0 && selection >= suffixPosition) {
+        return selection > suffixPosition ? suffixPosition
+          : this.delete || this.backspace ? suffixPosition : suffixPosition - 1
+      }
+
+      // Handle abs(value) < 10
+      // Make the caret stay where it were when the
+      // length of input is the same before and after
+      return this.backspace ? this.numeralCaretOnBackspace(selection, text)
+        : this.delete ? this.numeralCaretOnDelete(selection, text)
+          : this.numeralCaretOnInsert(selection)
+    },
+    numeralCaretOnInsert (selection) {
+      selection -= this.selection < selection ? 1 : 0
+      selection -= this.selection === selection ? 1 : 0
+
+      return selection
+    },
+    numeralCaretOnDelete (selection, text) {
+      if (text.substr(this.selection, 1) === this.decimal) {
```

Comment:
`text[this.selection]` (and few more similar occurences below and in `numeralCaretOnBackspace`)